src/main/java/seedu/address/model/person/predicates/SupplierPredicate.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.person.predicates;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Abstract class for predicates that operate on a {@code Person}.
+ */
+public abstract class SupplierPredicate implements Predicate<Person> {
```

Comment:
Well done. I think we can implement the same structure of having an abstract Predicate class for the delivery predicates as well.

Reply:
Yup I think this is a brilliant idea! Will implement that for the delivery predicate used for upcoming command as well in the next iteration